tests/network/test_continue_request.py

```diff
@@ -0,0 +1,241 @@
+#  Copyright 2023 Google LLC.
+#  Copyright (c) Microsoft Corporation.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+from typing import Literal
+
+import pytest
+from anys import ANY_DICT, ANY_STR
+from test_helpers import (ANY_TIMESTAMP, execute_command, send_JSON_command,
+                          subscribe, wait_for_event)
+
+
+@pytest.mark.asyncio
+async def test_continue_request_non_existent_request(websocket):
+    with pytest.raises(Exception) as exception_info:
+        await execute_command(
+            websocket, {
+                "method": "network.continueRequest",
+                "params": {
+                    "request": '_UNKNOWN_',
+                },
+            })
+    assert {
+        "error": "no such request",
+        "message": "No blocked request found for network id '_UNKNOWN_'"
+    } == exception_info.value.args[0]
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize("phase", [
+    "responseStarted",
+    pytest.param("authRequired",
+                 marks=pytest.mark.xfail(
+                     reason='TODO: #644: implement "authRequired" phase'))
+])
+async def test_continue_request_invalid_phase(websocket, context_id, phase):
+    # TODO: make offline.
+    url = "https://www.example.com/"
```

Comment:
NIT: if we move `https://www.example.com/` into a fixture, switching to local server will be way easier.

Reply:
This is a good point, but since this PR is getting quite big already, let's address it in a separate one.